Building Thrill as a dependency fails

I was trying to build the examples from the tutorial but in it's own project with Thrill as a dependency. So I added Thrill as a git submodule in my `lib` folder and created a very simple `CMakeLists.txt` like this:

```cmake
cmake_minimum_required (VERSION 2.6)
project (KMeans)
include_directories("lib/thrill")
add_subdirectory (lib/thrill)
set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O3 -DNDEBUG -std=c++14 -Wall -Wextra")
add_executable(KMeans k_means.cpp)
target_link_libraries(KMeans thrill)
```

When building the project I get an error:
```
[  5%] Built target s3
[  5%] Building CXX object lib/thrill/thrill/CMakeFiles/thrill.dir/vfs/s3_file.cpp.o
[...]/thrill_tut/lib/thrill/thrill/vfs/s3_file.cpp:18:19: fatal error: libs3.h: No such file or directory
compilation terminated.
lib/thrill/thrill/CMakeFiles/thrill.dir/build.make:1358: recipe for target 'lib/thrill/thrill/CMakeFiles/thrill.dir/vfs/s3_file.cpp.o' failed
make[2]: *** [lib/thrill/thrill/CMakeFiles/thrill.dir/vfs/s3_file.cpp.o] Error 1
CMakeFiles/Makefile2:1235: recipe for target 'lib/thrill/thrill/CMakeFiles/thrill.dir/all' failed
make[1]: *** [lib/thrill/thrill/CMakeFiles/thrill.dir/all] Error 2
Makefile:83: recipe for target 'all' failed
make: *** [all] Error 2
```

Maybe I'm missing something terribly stupid because I suck at CMake but in any case it would be great to have an example in the tutorial of how to build and include thrill as a lib.

Hmm, cmake is very "hacky". We havent put too much thought into how to use thrill as a library yet, but what you did looks right. 
"[...]/thrill_tut/lib/thrill/thrill/vfs/s3_file.cpp:18:19: fatal error: libs3.h: No such file or directory"
This is probably just a simple include_directories() to libs3 which is messed up in our CMakeLists.txt
Maybe you can fix it and send a pull request.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -426,7 +426,7 @@ if(THRILL_USE_S3 AND CURL_FOUND AND LIBXML2_FOUND AND OPENSSL_FOUND)
   # add libS3 and dependencies to thrill's dependencies
   set(THRILL_DEP_LIBRARIES ${THRILL_DEP_LIBRARIES} s3)
   add_definitions(-DTHRILL_HAVE_LIBS3=1)
-  include_directories("${CMAKE_SOURCE_DIR}/extlib/libs3/libs3/inc/")
+  include_directories("extlib/libs3/libs3/inc/")
 
   # saved code to detect libs3 on the system. use this once multipart upload is
   # mainstream.

